hw/ip/spi_device/doc/_index.md

```diff
@@ -309,6 +361,252 @@ All reads and writes to/from the SRAM for RXF and TXF activity are managed by
+BUSY bit is set by HW when it receives any commands that are uploaded to the FIFOs.
+SW may clear BUSY bit when it completes the received commands (e.g Erase/ Program).
+
+If the host sends the Write Status commands, the commands are not processed in this module.
+SW must configure the remaining command information entries to upload the Write Status commands to the FIFOs.
+
+### JEDEC ID Control
+
+JEDEC module returns JEDEC Device ID and Manufacturer ID following the Continuation Code (CC).
+SW may configure {{<regref "JEDEC_CC">}} CSR for HW to return proper CC.
+The *cc* field in {{<regref "JEDEC_CC">}} defines the return value, which is `0x7F` by default.
+*num_cc* defines how many times the HW to send CC byte before sending the JEDEC ID.
+
+The actual JEDEC ID consists of one byte device ID and two bytes manufacturer ID.
+The HW sends the device ID first, then `[7:0]` of manufacturer ID then `[15:8]` byte.
+
+### Serial Flash Discoverable Parameters (SFDP) Control
+
+HW parses SFDP command then fetch the data from SFDP space in the DPSRAM.
+HW provides 256B SFDP space.
+HW uses lower 8bit of the received 24 bit address to access the DPSRAM.
+Upper 16 bits are ignored (aliased).
+SW should prepare proper SFDP contents before the host system issues SFDP commands.
+
+HW fetches from the DPSRAM in 4B and returns the data to the SPI line.
+HW repeats the operation until CSb is de-asserted.
+
+### Read Command Processor
+
+The read command block has multiple sub-blocks to process normal Read, Fast Read, Fast Read Dual/ Quad from the internal DPSRAM.
+The DPSRAM has a 2kB region for the read command access.
+The read command region has two 1kB buffers.
+If HW recives the read access to the other half of the space first time, then the HW reports to the SW to refill the current 1kB region with new content.
+
+The double buffering scheme aids the SW to prepare the next chunk of data.
+SW copies a portion of data (1kB) from the internal flash memory into SPI_DEVICE DPSRAM.
+From the host system, the emulated SPI Device is seen more than 2kB storage device with the double buffering scheme.
+The assumption is that the host system reads mostly sequentially.
+
+#### Address Handling
+
+For read commands such as Normal Read, Fast Read {Single/ Dual/ Quad} Output commands, the address comes through ID0 only.
+The state machine in this block shifts the address one-by-one and decrements the address counter register by 1.
+
+When it reaches the 4B address (`addr[2]`), the module triggers the DPSRAM state machine to fetch data from the DPSRAM.
+When the module receives `addr[0]`, at the positive edge of SCK, the module moves to appropriate command state based on the given CMD_INFO data.
+
+If the received address falls into mailbox address ragne and mailbox feature is enabled, the module turns on the mailbox selection bit.
+Then all out-going requests to the DPSRAM are forwarded to the mailbox section, not the read buffer section.
```

Comment:
just wanted to clarify one thing here. If the mailbox space is for example located ahead of other space, we DO NOT dynamically switch right? Ie, once you start in mailbox, even if you read into non-mailbox space, will we still remain in mailbox space? 

I'm not suggesting we change anything, just that it would be good to clarify the behavior.

Reply:
Hum... good point. Let me check the design. I am honestly not sure about the behavior.